clang/lib/CodeGen/CGLoopInfo.cpp

```diff
@@ -524,6 +524,48 @@ MDNode *LoopInfo::createMetadata(
     LoopProperties.push_back(MDNode::get(Ctx, Vals));
   }
 
+  if (Attrs.SYCLLoopCoalesceEnable) {
+    LLVMContext &Ctx = Header->getContext();
```

Comment:
The `Ctx` extraction **by reference** seems to be taking place under every condition. Is a single prior call to `getContext()` possible in the outer scope?

Reply:
Moreover, it is already initialized at line 502.